test/mobile/op_deps/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(op_deps_project)
+
+# Find torch library
+add_library(torch UNKNOWN IMPORTED)
```

Comment:
For my information: how come you need this line before the find_package? I would have expected find_package to produce the library on its own without extra prompting

Reply:
For whatever reason it only searches for "Caffe2" cmake package when it's building shared lib:
https://github.com/pytorch/pytorch/blob/master/cmake/TorchConfig.cmake.in#L40
And "torch" target is defined in caffe2/CMakeLists.txt:
https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L642

So it seems to me the static library isn't properly supported as you mentioned: #21737
I don't know how involved it might be to fix it so figured this workaround...

Reply:
Looks like TorchConfig.cmake needs some updating. Can you file a bug about this? Seems worth fixing.

Reply:
Sure, I just made an attempt to fix this problem: #29837. Let's see if it can pass all CIs.